frontend/src/concepts/pipelines/content/pipelinesDetails/PipelineDetails.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { useParams } from 'react-router-dom';
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  Drawer,
+  DrawerContent,
+  DrawerContentBody,
+  Tab,
+  TabContent,
+  Tabs,
+  TabTitleText,
+} from '@patternfly/react-core';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import usePipelineTemplate from '~/concepts/pipelines/apiHooks/usePipelineTemplate';
+import { PipelineTopology, usePipelineTaskTopology } from '~/concepts/pipelines/topology';
+import usePipelineById from '~/concepts/pipelines/apiHooks/usePipelineById';
+import MarkdownView from '~/components/MarkdownView';
+import SelectedTaskDrawerContent from '~/concepts/pipelines/content/pipelinesDetails/SelectedTaskDrawerContent';
+import PipelineDetailsActions from '~/concepts/pipelines/content/pipelinesDetails/PipelineDetailsActions';
+import PipelineDetailsYAML from '~/concepts/pipelines/content/pipelinesDetails/PipelineDetailsYAML';
+
+type PipelineDetailsProps = {
+  breadcrumbPath: React.ReactElement<typeof BreadcrumbItem>[];
+};
+
+enum PipelineDetailsTab {
+  GRAPH,
+  YAML,
+}
+
+const PipelineDetails: React.FC<PipelineDetailsProps> = ({ breadcrumbPath }) => {
+  const { pipelineId } = useParams();
+  const [pipeline, pipelineLoad, pipelineLoadError] = usePipelineById(pipelineId);
+  const [pipelineRun, pipelineTemplateLoaded, templateLoadError] = usePipelineTemplate(pipelineId);
+  const [activeTabKey, setActiveTabKey] = React.useState<string | number>(PipelineDetailsTab.GRAPH);
+  const [selectedId, setSelectedId] = React.useState<string | null>(null);
+  const { taskMap, nodes } = usePipelineTaskTopology(pipelineRun);
+
+  return (
+    <Drawer isExpanded={!!selectedId}>
```

Comment:
This render is large -- I'd like to reduce it, but I'm having all kinds of styling issues with containers not filling. I will look to cleanup in the future when the solution for how the styling will work is figured out.